### PR TITLE
Gateway docs update

### PR DESCRIPTION
**Pull request by okjodom**
Cherry picked the documentation parts of #1454 .
These doc refresh add detail on the components that make up the new gateway

### CODE REVIEW

File: docs/gateway.md [hunk 1]
```diff
@@ -1,49 +1,54 @@
...
-      Component(an, "Gateway Actor N", "Serving Federation N", "An actor component used by the gateway to serve Federation N")
-    }
-  }
+### Gateway-lnrpc-extension
 
-  Container_Boundary(c, "Federations served by the Gateway") {
-    Container(f1, "Federation 1", "Fedeimint Federation", "Fedimint Federation providing custody to multiple users")
-    Container(fn, "Federation N", "Fedeimint Federation", "Fedimint Federation providing custody to multiple users")
-  }
+A Lightning extension / plugin service that provides all the necessary Lightning functionalities the gateway webserver daemon.
```
Comment (elsirion) [inline docs/gateway.md:32]: This sentence seems to miss a word/is incomplete?

**Review by elsirion — APPROVED** [review]
Looks good enough, I don't want to stall this, docs are a continuous effort anyway.

**Comment by mxz42** [comment]
nice documentation @okjodom 

> Looks good enough, I don't want to stall this, docs are a continuous effort anyway.

I have a slightly different opinion on this. First off, I agree, documentation is a continuous effort. 
But it's very unlikely that the documentation PR will conflict with anything and a rebase will be necessary. 

I do have a couple nits here and there.

- It's easier to point them out in a current PR, even with possible suggestions attached instead of looking for the markdown file and creating a new PR
- easier to have a quick github conversation about wording sentence or clarifying things
- having this PR open for a couple days gives more people the chance to look at it and read it. Again, here it's easier to chime in with a comment instead of looking for the md file, actually coming up with a change, creating a commit, pushing it to github, and create the PR. 


TLDR: I don't think it hurts to have documentation PR's open for a couple days to give more people the chance to look at it as it's unlikely that an expensive rebase will be necessary. 

btw, I'm always happy to proof read/review documentation, so just tag me.
